In [3]:
import pandas as pd

In [4]:
# cite_seq = pd.read_hdf('./data/EDA/test_cite_inputs.h5')
cite_seq = pd.read_hdf('./data/EDA/train_cite_inputs.h5')
cite_seq.head()

gene_id       ENSG00000121410_A1BG  ENSG00000268895_A1BG-AS1  \
cell_id                                                        
45006fe3e4c8                   0.0                       0.0   
d02759a80ba2                   0.0                       0.0   
c016c6b0efa5                   0.0                       0.0   
ba7f733a4f75                   0.0                       0.0   
fbcf2443ffb2                   0.0                       0.0   

gene_id       ENSG00000175899_A2M  ENSG00000245105_A2M-AS1  \
cell_id                                                      
45006fe3e4c8                  0.0                      0.0   
d02759a80ba2                  0.0                      0.0   
c016c6b0efa5                  0.0                      0.0   
ba7f733a4f75                  0.0                      0.0   
fbcf2443ffb2                  0.0                      0.0   

gene_id       ENSG00000166535_A2ML1  ENSG00000128274_A4GALT  \
cell_id                                                       
45006fe3e4c8                    0.0                0.000000   
d02759a80ba2                    0.0                0.000000   
c016c6b0efa5                    0.0                3.847321   
ba7f733a4f75                    0.0                0.000000   
fbcf2443ffb2                    0.0                0.000000   

gene_id       ENSG00000094914_AAAS  ENSG00000081760_AACS  \
cell_id                                                    
45006fe3e4c8              0.000000              0.000000   
d02759a80ba2              0.000000              0.000000   
c016c6b0efa5              0.000000              3.847321   
ba7f733a4f75              3.436846              3.436846   
fbcf2443ffb2              0.000000              4.196826   

gene_id       ENSG00000109576_AADAT  ENSG00000103591_AAGAB  ...  \
cell_id                                                     ...   
45006fe3e4c8               0.000000                    0.0  ...   
d02759a80ba2               0.000000                    0.0  ...   
c016c6b0efa5               3.847321                    0.0  ...   
ba7f733a4f75               0.000000                    0.0  ...   
fbcf2443ffb2               0.000000                    0.0  ...   

gene_id       ENSG00000153975_ZUP1  ENSG00000086827_ZW10  \
cell_id                                                    
45006fe3e4c8              0.000000              0.000000   
d02759a80ba2              0.000000              0.000000   
c016c6b0efa5              0.000000              0.000000   
ba7f733a4f75              3.436846              0.000000   
fbcf2443ffb2              0.000000              4.196826   

gene_id       ENSG00000174442_ZWILCH  ENSG00000122952_ZWINT  \
cell_id                                                       
45006fe3e4c8                0.000000               0.000000   
d02759a80ba2                0.000000               4.039545   
c016c6b0efa5                3.847321               4.529743   
ba7f733a4f75                4.113780               5.020215   
fbcf2443ffb2                4.196826               4.196826   

gene_id       ENSG00000198205_ZXDA  ENSG00000198455_ZXDB  \
cell_id                                                    
45006fe3e4c8                   0.0                   0.0   
d02759a80ba2                   0.0                   0.0   
c016c6b0efa5                   0.0                   0.0   
ba7f733a4f75                   0.0                   0.0   
fbcf2443ffb2                   0.0                   0.0   

gene_id       ENSG00000070476_ZXDC  ENSG00000162378_ZYG11B  \
cell_id                                                      
45006fe3e4c8               0.00000                0.000000   
d02759a80ba2               0.00000                0.000000   
c016c6b0efa5               0.00000                3.847321   
ba7f733a4f75               0.00000                3.436846   
fbcf2443ffb2               3.51861                4.196826   

gene_id       ENSG00000159840_ZYX  ENSG00000074755_ZZEF1  
cell_

In [5]:
cite_seq.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70988 entries, 45006fe3e4c8 to c91b6b2ccd3d
Columns: 22050 entries, ENSG00000121410_A1BG to ENSG00000074755_ZZEF1
dtypes: float32(22050)
memory usage: 5.8+ GB


RNA library-size normalized* and log1p transformed* counts (gene expression levels), with 48,663 rows corresponding to cells and 22,050 columns corresponding to genes given by {gene_name}_{gene_ensemble-ids}.

Library sized normalization: normalize so all cells have the same total gene count. TODO: understand cite-seq assay better.

Some confusion about this in forums:
- https://www.kaggle.com/competitions/open-problems-multimodal/discussion/346341

In [6]:
cite_seq.head().T.sum()

cell_id
45006fe3e4c8    18104.728516
d02759a80ba2    17466.855469
c016c6b0efa5    20732.220703
ba7f733a4f75    25030.843750
fbcf2443ffb2    25476.197266
dtype: float32

In [7]:
cite_seq.describe()

gene_id  ENSG00000121410_A1BG  ENSG00000268895_A1BG-AS1  ENSG00000175899_A2M  \
count            70988.000000              70988.000000         70988.000000   
mean                 0.086048                  0.199110             0.017556   
std                  0.564659                  0.843376             0.263296   
min                  0.000000                  0.000000             0.000000   
25%                  0.000000                  0.000000             0.000000   
50%                  0.000000                  0.000000             0.000000   
75%                  0.000000                  0.000000             0.000000   
max                  5.402543                  5.697094             6.385898   

gene_id  ENSG00000245105_A2M-AS1  ENSG00000166535_A2ML1  \
count               70988.000000           70988.000000   
mean                    0.167338               0.001346   
std                     0.783140               0.071854   
min                     0.000000               0.000000   
25%                     0.000000               0.000000   
50%                     0.000000               0.000000   
75%                     0.000000               0.000000   
max                     5.565943               4.376676   

gene_id  ENSG00000128274_A4GALT  ENSG00000094914_AAAS  ENSG00000081760_AACS  \
count              70988.000000          70988.000000          70988.000000   
mean                   0.096629              1.732955              1.640168   
std                    0.596086              1.977421              1.982317   
min                    0.000000              0.000000              0.000000   
25%                    0.000000              0.000000              0.000000   
50%                    0.000000              0.000000              0.000000   
75%                    0.000000              3.816849              3.788277   
max                    5.574059              5.898176              6.218979   

gene_id  ENSG00000109576_AADAT  ENSG00000103591_AAGAB  ...  \
count             70988.000000           70988.000000  ...   
mean                  0.286222               1.276829  ...   
std                   1.003846               1.829827  ...   
min                   0.000000               0.000000  ...   
25%                   0.000000               0.000000  ...   
50%                   0.000000               0.000000  ...   
75%                   0.000000               3.530414  ...   
max                   5.365835               5.748921  ...   

gene_id  ENSG00000153975_ZUP1  ENSG00000086827_ZW10  ENSG00000174442_ZWILCH  \
count            70988.000000          70988.000000            70988.000000   
mean                 0.993167              1.458712                1.712961   
std                  1.685713              1.900358                1.978515   
min                  0.000000              0.000000                0.000000   
25%                  0.000000              0.000000                0.000000   
50%                  0.000000              0.000000                0.000000   
75%                  3.191627              3.652694                3.800253   
max                  5.727450              5.811031                5.905997   

gene_id  ENSG00000122952_ZWINT  ENSG00000198205_ZXDA  ENSG00000198455_ZXDB  \
count             70988.000000          70988.000000          70988.000000   
mean                  3.404964              0.059131              0.200200   
std                   2.073926              0.467184              0.842347   
min                   0.000000              0.000000              0.000000   
25%                   0.000000              0.000000              0.000000   
50%                   4.307628              0.000000              0.000000   
75%                   4.885360              0.000000              0.000000   
max                   6.550168              5.219127              5.033606   

gene_id  ENSG00000070476_ZXDC  ENSG00000162378_ZYG11B  ENSG00

In [8]:
# across genes, expression is usually low across cells
cite_seq.describe().T.mean()

count    70988.000000
mean         0.977238
std          0.995665
min          0.015569
25%          0.393084
50%          0.769719
75%          1.404981
max          5.533523
dtype: float64

In [4]:
cite_seq_prot = pd.read_hdf('./data/EDA/train_cite_targets.h5')
cite_seq_prot.head()

gene_id           CD86     CD274     CD270     CD155     CD112      CD47  \
cell_id                                                                    
45006fe3e4c8  1.167804  0.622530  0.106959  0.324989  3.331674  6.426002   
d02759a80ba2  0.818970  0.506009  1.078682  6.848758  3.524885  5.279456   
c016c6b0efa5 -0.356703 -0.422261 -0.824493  1.137495  0.518924  7.221962   
ba7f733a4f75 -1.201507  0.149115  2.022468  6.021595  7.258670  2.792436   
fbcf2443ffb2 -0.100404  0.697461  0.625836 -0.298404  1.369898  3.254521   

gene_id            CD48      CD40     CD154      CD52  ...      CD94  \
cell_id                                                ...             
45006fe3e4c8   1.480766 -0.728392 -0.468851 -0.073285  ... -0.448390   
d02759a80ba2   4.930438  2.069372  0.333652 -0.468088  ...  0.323613   
c016c6b0efa5  -0.375034  1.738071  0.142919 -0.971460  ...  1.348692   
ba7f733a4f75  21.708519 -0.137913  1.649969 -0.754680  ...  1.504426   
fbcf2443ffb2  -1.659380  0.643531  0.902710  1.291877  ...  0.777023   

gene_id           CD162     CD85j      CD23     CD328     HLA-E      CD82  \
cell_id                                                                     
45006fe3e4c8   3.220174 -0.533004  0.674956 -0.006187  0.682148  1.398105   
d02759a80ba2   8.407108  0.131301  0.047607 -0.243628  0.547864  1.832587   
c016c6b0efa5   4.888579 -0.279483 -0.131097 -0.177604 -0.689188  9.013709   
ba7f733a4f75  12.391979  0.511394  0.587863 -0.752638  1.714851  3.893782   
fbcf2443ffb2   6.496499  0.279898 -0.841950 -0.869419  0.675092  5.259685   

gene_id          CD101      CD88     CD224  
cell_id                                     
45006fe3e4c8  0.414292  1.780314  0.548070  
d02759a80ba2  0.982308  2.736507  2.184063  
c016c6b0efa5 -1.182975  3.958148  2.868600  
ba7f733a4f75  1.799661  1.537249  4.407671  
fbcf2443ffb2 -0.835379  9.631781  1.765445  

[5 rows x 140 columns]

In [5]:
cite_seq_prot.describe()

gene_id          CD86         CD274         CD270         CD155         CD112  \
count    70988.000000  70988.000000  70988.000000  70988.000000  70988.000000   
mean         0.454768      0.466369      0.918049      5.610967      5.307707   
std          1.393927      0.894457      0.923267      2.542798      2.043094   
min         -5.717155     -5.292869     -4.187119     -3.907870     -3.107560   
25%         -0.366943     -0.144443      0.289499      3.830644      3.961191   
50%         -0.031319      0.416705      0.892687      5.395159      5.324531   
75%          1.050250      1.014566      1.519067      7.176124      6.682766   
max         37.034649      8.832175      6.191772     18.115868     14.311632   

gene_id          CD47          CD48          CD40         CD154          CD52  \
count    70988.000000  70988.000000  70988.000000  70988.000000  70988.000000   
mean         7.447880      4.052779      0.399293      0.714440      0.840576   
std          3.593255      5.332760      1.055089      0.858400      1.245953   
min         -4.439901     -8.152050     -4.893185     -4.166430     -3.502274   
25%          4.861448      0.257704     -0.315521      0.144492     -0.028824   
50%          7.269966      2.225405      0.302452      0.696771      0.694199   
75%          9.791561      6.408164      0.987647      1.268129      1.524002   
max         27.746946     44.257801     22.913204      4.842474     13.358733   

gene_id  ...          CD94         CD162         CD85j          CD23  \
count    ...  70988.000000  70988.000000  70988.000000  70988.000000   
mean     ...      0.269132      7.229657      0.871969      0.240155   
std      ...      0.835048      3.189166      1.135966      0.736558   
min      ...     -6.492634     -9.677796     -6.471857     -7.369277   
25%      ...     -0.294844      5.033357      0.070054     -0.231999   
50%      ...      0.259515      7.037832      0.762008      0.237440   
75%      ...      0.815943      9.227937      1.535960      0.703372   
max      ...      5.813220     23.441250     11.292436      5.764631   

gene_id         CD328         HLA-E          CD82         CD101          CD88  \
count    70988.000000  70988.000000  70988.000000  70988.000000  70988.000000   
mean         0.236248      0.777902      5.375080      0.629204      3.608978   
std          1.801219      0.943353      2.876905      1.363843      2.722020   
min         -5.951009     -2.352817    -14.125098     -2.993998    -29.403835   
25%         -0.286252      0.123200      3.295399     -0.110286      1.666111   
50%         -0.121567      0.743739      5.057893      0.474196      2.945657   
75%          0.040861      1.386421      7.130883      1.124541      5.183252   
max         77.956345      7.225060     22.147051     26.483639     14.911215   

gene_id         CD224  
count    70988.000000  
mean         3.418748  
std          2.465487  
min         -4.654044  
25%          1.606031  
50%          3.011245  
75%          4.826133  
max         18.671375  

[8 rows x 140 columns]

In [6]:
cite_seq_prot.describe().T.mean()

count    70988.000000
mean         1.886106
std          1.759287
min         -6.544227
25%          0.696609
50%          1.616853
75%          2.791192
max         21.659718
dtype: float64